core/slice.go

```diff
@@ -179,15 +176,13 @@ func (sl *Slice) Append(header *types.Header, domPendingHeader *types.Header, do
 		return nil, err
 	}
 
-	if !domOrigin {
-		// CalcTd on the new block
-		td, err = sl.calcTd(block.Header())
-		if err != nil {
-			return nil, err
-		}
-		// HLCR
-		reorg = sl.hlcr(td)
+	// CalcTd on the new block
+	td, err := sl.calcTd(block.Header(), domTd)
+	if err != nil {
+		return nil, err
 	}
+	// HLCR
+	reorg = sl.hlcr(td)
```

Comment:
I may be wrong, but IIUC, this decision will cause the local head to override the dom head if local TD > dom TD.

Reply:
Yes this is the case, and is intended.

Reply:
This comment is out of date. When I left it, the code violated HLCR. The code has since been modified to follow HLCR, and my original comment no longer stands :+1: